### PR TITLE
Ports Multilingual quirk from Beestation, makes it not random.

### PR DESCRIPTION
**Pull request by Ms-Mee**
## About The Pull Request

This PR ports https://github.com/BeeStation/BeeStation-Hornet/pull/2228, additionally making the language not random and instead selectable.
Additional credits go to super12pl for the initial port and attempt at adding such a feature, and AsciiSquid who tested and fixed the language selection button, which I could not.

## Why It's Good For The Game

Being able to select a secondary language is a nice and simple way of improving character customization, and this makes the language system overall a bit more present.

## Changelog
:cl: DatBoiTim, yorii, super12pl, Ms-Mee, AsciiSquid
add: Added a new quirk which allows you to select a second language your character understands and speaks.
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -87,6 +87,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/list/randomise = list(RANDOM_UNDERWEAR = TRUE, RANDOM_UNDERWEAR_COLOR = TRUE, RANDOM_UNDERSHIRT = TRUE, RANDOM_SOCKS = TRUE, RANDOM_BACKPACK = TRUE, RANDOM_JUMPSUIT_STYLE = TRUE, RANDOM_EXOWEAR_STYLE = TRUE, RANDOM_HAIRSTYLE = TRUE, RANDOM_HAIR_COLOR = TRUE, RANDOM_FACIAL_HAIRSTYLE = TRUE, RANDOM_FACIAL_HAIR_COLOR = TRUE, RANDOM_SKIN_TONE = TRUE, RANDOM_EYE_COLOR = TRUE)
 	var/list/friendlyGenders = list("Male" = "male", "Female" = "female", "Other" = "plural")
 	var/phobia = "spiders"
+	var/language = "Kalixcian Common"
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]: Doesn't need to exist.

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					if(phobiaType)
 						phobia = phobiaType
 
+				if("language")
+					var/obj/item/organ/tongue/T = new pref_species.mutanttongue
+					var/list/languages_possible = T.languages_possible
+					var/datum/language_holder/language_holder = new pref_species.species_language_holder(T)
+					languages_possible -= typecacheof(/datum/language/codespeak)
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
					languages_possible -= GLOB.restricted_languages
```

make a global list for restricted roundstart languages

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+					languages_possible -= language_holder.understood_languages
+					languages_possible -= language_holder.spoken_languages
+					languages_possible -= language_holder.blocked_languages
+					//Credit To Yowii/Yoworii/Yorii for a much more streamlined method of language library building
+					var/list/language_names = list()
+					for(var/L in languages_possible)
+						var/datum/language/lang_possible = L
+						language_names |= initial(lang_possible.name)
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
					var/list/language_options = list()
					for(var/datum/language/lang_option as anything in languages_possible)
						language_options |= initial(lang_option.name)
						language_options[initial(lang_option.name)] = lang_option
```

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+					var/languageType = input(user, "What other language do you know?", "Character Preference", language) as null|anything in language_names
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
					var/selected_language= input(user, "What other language do you know?", "Character Preference", language) as null|anything in language_options
```

Comment (Kapu1178) [inline code/modules/client/preferences.dm:1969]:
```suggestion
					var/selected_language = input(user, "What other language do you know?", "Character Preference", language) as null|anything in language_names
```

File: code/modules/client/preferences.dm [hunk 5]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+					if(languageType)
+						language = languageType
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
					if(selected_language)
```

File: code/modules/client/preferences.dm [hunk 6]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+						for(var/datum/language/language_datum_test as anything in GLOB.all_languages)
+							if(initial(language_datum_test.name) == language)
+								language_datum = language_datum_test
+								break
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
						language_datum = language_options[selected_language]
```

File: code/modules/client/preferences.dm [hunk 7]
```diff
@@ -1946,6 +1953,27 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+
 				if ("max_chat_length")
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm:1976]:
```suggestion
					qdel(language_holder)
					qdel(T)

				if ("max_chat_length")
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/language.dm [hunk 8]
```diff
@@ -14,5 +14,5 @@ SUBSYSTEM_DEF(language)
 		var/datum/language/instance = new language
 
 		GLOB.language_datum_instances[language] = instance
-
+	GLOB.restricted_languages = list(/datum/language/codespeak)
```
Comment (Kapu1178) [inline code/controllers/subsystem/language.dm]:
```suggestion
	GLOB.restricted_languages += /datum/language/codespeak
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by Ms-Mee** [comment]
For some reason changing the language is making it so you can't speak any other language
Loading the savefile works perfectly fine
So probably something in the selection part is causing that
What? I have no clue...

**Pull request closed without merging**